subprojects/core/src/main/java/org/gradle/api/Task.java

```diff
@@ -359,6 +359,19 @@ public String determineName(Task c) {
     Task doFirst(Closure action);
 
     /**
+     * <p>Adds the given closure to the beginning of this task's action list. The closure is passed this task as a
+     * parameter when executed.</p>
+     *
+     * @param actionName An arbitrary string that is used for logging.
+     * @param action The action closure to execute.
+     * @return This task.
+     *
+     * @since 4.2
+     */
+    @Incubating
+    Task doFirst(String actionName, Closure action);
```

Comment:
I think it's not worth it since it would cause deprecation warnings on practically every project and if we ever removed these methods, we'd probably also mix in something to restore them (to keep old plugins working).